### PR TITLE
Analysis UI Components + First Derivative Plot

### PR DESCRIPTION
**Pull request by adhampton110**
**Overview**
-Added user input for the "blank sample" of the dataset.
-UI components were added to the "Analysis" tab.
-User is able to view the raw data or both the raw data along with the First Derivative for each respective sample. This sketch is generated using polynomial regression and is referenced by using MeltR's pre-existing functions. 
-The viewing option (view raw data or raw data with first derivative) is made optional by the use of a checkbox contained in a side panel. 

**Demonstration**
-Here are a couple of screenshots demonstrating my changes...

**Added "Blank" Text Box**
blank

**Screenshot highlighting background subtraction (Whichever sample is listed as the blank, is automatically omitted)**
 
background sub

**Raw Data Output**
raw

**First Derivative Plotted Over Raw Data**
first + raw

### CODE REVIEW

File: server.r [hunk 1]
```diff
@@ -52,37 +52,89 @@ server <- function(input,output, session){
...
+        #plot containing first derivative with raw data
+        plotDerivative = paste0("plotDerivative",myI)
+        output[[plotDerivative]] <- renderPlot({
+          data = values$masterFrame[values$masterFrame$Sample == myI,]
+          data %>%
+            arrange(Temperature) %>%
+            mutate(slope = (Absorbance - lag(Absorbance)) /
+                     (Temperature - lag(Temperature))) %>%
+            ggplot(aes(Temperature)) +
+            geom_line(aes(y = Absorbance), size = 1.2) +
```
Comment (aithaprasad) [inline server.r:131]: No magic numbers please. All numbers should go on top of the file as a variable and then use the variable wherever required

File: server.r [hunk 2]
```diff
@@ -52,37 +52,89 @@ server <- function(input,output, session){
...
+            geom_smooth(aes(y = slope * 20 + 1.4), se = FALSE, size = 0.8) +
```
Comment (aithaprasad) [inline server.r:132]: same here

File: server.r (hunk 1 above)
Comment (adhampton110) [reply server.r:131]: Understood, will go back & refine.

Comment (adhampton110) [reply server.r]: Please checked my revised implementation of my first derivative sketch.

File: server.r [hunk 3]
```diff
@@ -53,42 +66,86 @@ server <- function(input,output, session){
...
-        output[[plotName]] <- renderPlot({
-          data = values$masterFrame[values$masterFrame$Sample == myI,]
-          ggplot(data, aes(x = Temperature, 
-                           y = Absorbance, 
-                           color = factor(Sample))) +
-            geom_point() + theme_classic() +
-            geom_vline(xintercept = input[[plotSlider]][1]) +
-            geom_vline(xintercept = input[[plotSlider]][2])
-        })
+      #if i != blank
```
Comment (aithaprasad) [inline server.r:128]: If this is not necessary for us in the future. Please remove it.

Comment (adhampton110) [reply server.r:128]: Fixed hardcoded line to account for the blank selected by the user.

**Review by aithaprasad — APPROVED** [review]

**Pull request merged**